squire_lib/src/ffi_tournament.rs

```diff
@@ -53,41 +57,75 @@ impl TournamentId {
         return ptr;
     }
 
+    /// Adds a player to a tournament
+    /// On error a NULL UUID is returned
+    #[no_mangle]
+    pub unsafe extern "C" fn tid_add_player(self: Self, __name: *const c_char) -> PlayerId {
+        let name: &str = unsafe { CStr::from_ptr(__name).to_str().unwrap() };
+        let op: TournOp =
+            TournOp::RegisterPlayer(SquireAccount::new(name.to_string(), name.to_string()));
+
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get_mut(&self) {
+            Some(mut t) => {
+                match t.apply_op(op) {
+                    Ok(RegisterPlayer(PlayerIdentifier::Id(id))) => {
+                        return id;
+                    }
+                    Err(t_err) => {
+                        println!("[FFI]: {t_err}");
+                        return PlayerId::new(Uuid::from_bytes(NULL_UUID_BYTES));
+                    }
+                    // We are known the from that the data will take if it exists
+                    // so we can ignore the other outcomes
+                    _ => {
+                        return Uuid::default().into();
+                    }
+                };
+            }
+            None => {
+                println!("[FFI]: Cannot find tournament in tid_add_player");
+                return PlayerId::new(Uuid::from_bytes(NULL_UUID_BYTES));
+            }
+        }
+    }
+
     /// Returns the name of a tournament
     /// Returns NULL if an error happens
     /// This is heap allocated, please free it
     #[no_mangle]
-    pub extern "C" fn tid_name(self: Self) -> *const c_char {
+    pub unsafe extern "C" fn tid_name(self: Self) -> *const c_char {
         let tourn: Tournament;
-        match FFI_TOURNAMENT_REGISTRY.get().unwrap().get(&self) {
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get(&self) {
             Some(t) => tourn = t.value().clone(),
             None => {
+                println!("[FFI]: Cannot find tournament");
                 return std::ptr::null();
             }
         }
-        return unsafe { clone_string_to_c_string(tourn.name) };
+        return clone_string_to_c_string(tourn.name);
     }
 
     /// Returns the format of a tournament
     /// Returns NULL if an error happens
     /// This is heap allocated, please free it
     #[no_mangle]
-    pub extern "C" fn tid_format(self: Self) -> *const c_char {
+    pub unsafe extern "C" fn tid_format(self: Self) -> *const c_char {
```

Comment:
Same comments as before.
```suggestion
    pub extern "C" fn tid_format(self: Self) -> *const c_char {
```